### PR TITLE
Add pl.suve.colorful

### PR DESCRIPTION
**Pull request by suve**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.

    _Colorful is a simple 2D side-view shooter game, where the player travels a maze of caves and corridors in search of colorful artefacts._

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. ~*(If not, please explain each non-standard permission.)*~
- [x] All assets referenced in the manifest are redistributable by any party. ~If not, the unredistributable parts are using an extra-data source type.~
- [x] I am an author/developer/upstream contributor of the project. ~If not, I contacted upstream developers about submitting their software to Flathub.~ 
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.

- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

    _The patch fixing file paths simply replaces hard-coded instances of `/usr` with hard-coded `/app`. I will fix this in the game's original repo by adding proper `PREFIX` support._

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: 0f7861669b9df34f4826f1a6085872d1a3000b5f.patch [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+--- a/Makefile.in
++++ b/Makefile.in
+@@ -48,8 +48,8 @@ install: all
+ 	install -m 644 -p -D "pkg/colorful-64x64.png"   "$(DESTDIR)/usr/share/icons/hicolor/64x64/apps/colorful.png"
+ 	install -m 644 -p -D "pkg/colorful-128x128.png" "$(DESTDIR)/usr/share/icons/hicolor/128x128/apps/colorful.png"
+ 	# Desktop file and AppStream metadata
+-	install -m 644 -p -D "pkg/colorful.desktop"     "$(DESTDIR)/usr/share/applications/colorful.desktop"
+-	install -m 644 -p -D "pkg/colorful.appdata.xml" "$(DESTDIR)/usr/share/appdata/colorful.appdata.xml"
++	install -m 644 -p -D "pkg/pl.suve.colorful.desktop"      "$(DESTDIR)/usr/share/applications/pl.suve.colorful.desktop"
++	install -m 644 -p -D "pkg/pl.suve.colorful.metainfo.xml" "$(DESTDIR)/usr/share/metainfo/pl.suve.colorful.metainfo.xml"
```
Comment (hfiguiere) [inline 0f7861669b9df34f4826f1a6085872d1a3000b5f.patch:25]: this is the wrong path too. `/usr` should be replaced by `$(PREFIX)`

for the name change you can put `rename-desktop-file` and `rename-appdata-file` in the toplevel of the manifest.

File: pl.suve.colorful.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+command: colorful
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --socket=pulseaudio
+  - --device=dri
+
+modules:
+  - name: optipng
```
Comment (hfiguiere) [inline pl.suve.colorful.yaml:25]: of this is build time only, a cleanup rule of `*` is missing.

File: pl.suve.colorful.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    sources:
+      - type: archive
+        url: https://netcologne.dl.sourceforge.net/project/optipng/OptiPNG/optipng-0.7.8/optipng-0.7.8.tar.gz
+        sha256: 25a3bd68481f21502ccaa0f4c13f84dcf6b20338e4c4e8c51f2cefbd8513398c
+    buildsystem: autotools
+    build-options:
+      config-opts: 
+        - --with-system-libs
+
+  - name: vorbis-tools
```
Comment (hfiguiere) [inline pl.suve.colorful.yaml:35]: of this is build time only, a cleanup rule of `*` is missing.

File: pl.suve.colorful.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      # Upstream commit that adds info about releases to metadata
+      # https://github.com/suve/LD25/commit/d94d992293aa77c9ae0a8b2c9019f51b5ef21e4b
+      - type: patch
+        path: d94d992293aa77c9ae0a8b2c9019f51b5ef21e4b.patch
+
+      # Makefile is hardcoded to /usr, switch to /app
+      - type: patch
+        path: fix-file-paths.patch
+
+    buildsystem: simple
```
Comment (hfiguiere) [inline pl.suve.colorful.yaml:67]: this should be `buildsystem: autotools`

File: pl.suve.colorful.yaml (hunk 2 above)
Comment (suve) [reply pl.suve.colorful.yaml:25]: Thanks, fixed in 5659b31a8bc1afbd02b30ac6a324af6067f28c77.

File: pl.suve.colorful.yaml (hunk 3 above)
Comment (suve) [reply pl.suve.colorful.yaml:37]: Fixed in 5659b31a8bc1afbd02b30ac6a324af6067f28c77.

**Comment by hfiguiere** [comment]
bot, build pl.suve.colorful

File: pl.suve.colorful.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
+app-id: pl.suve.colorful
```
[the remark this answers is not in the document]
Comment (bbhtt) [reply pl.suve.colorful.yaml:1]: Yes setting up a redirect or just a basic page with links to source repo and author will be nice.

[the remark this answers is not in the document]
Comment (suve) [reply pl.suve.colorful.yaml:1]: I replaced the "under construction" banner with a basic webpage containing some links to the repo and other downloads.

[the remark this answers is not in the document]
Comment (bbhtt) [reply pl.suve.colorful.yaml:1]: Looks good.

**Comment by bbhtt** [comment]
bot, build pl.suve.colorful

**Pull request closed without merging**